docs/source/contributor-guide/branching_model.md

```diff
@@ -0,0 +1,35 @@
+<!---
+  Licensed to the Apache Software Foundation (ASF) under one
+  or more contributor license agreements.  See the NOTICE file
+  distributed with this work for additional information
+  regarding copyright ownership.  The ASF licenses this file
+  to you under the Apache License, Version 2.0 (the
+  "License"); you may not use this file except in compliance
+  with the License.  You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing,
+  software distributed under the License is distributed on an
+  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+  KIND, either express or implied.  See the License for the
+  specific language governing permissions and limitations
+  under the License.
+-->
+
+# DataFusion Branching Model
+
+Most PRs are raised directly against master, and we keep master in a releasable state at all times by
+only having dependencies on released crates. This allows us to reliably release new versions of DataFusion
+on a fixed schedule.
+
+This is problematic when making breaking changes to dependencies such as `arrow-rs` or `sqlparser` and then making
```

Comment:
```suggestion
This is problematic when making changes to dependencies such as `arrow-rs` or `sqlparser` and then making
```

Reply:
e.g. I'm currently waiting on https://github.com/apache/arrow-rs/pull/2404 which was not a breaking change, was merged a while ago, but still hasn't made it into DataFusion or Ballista. As a result, we're maintaining our own fork of all 3 repos.